code/modules/power/singularity/containment_field.dm

```diff
@@ -14,12 +14,14 @@
 	can_atmos_pass = ATMOS_PASS_NO
 	light_range = 4
 	layer = ABOVE_OBJ_LAYER
+	explosion_block = INFINITY
 	///First of the generators producing the containment field
 	var/obj/machinery/field/generator/field_gen_1 = null
 	///Second of the generators producing the containment field
 	var/obj/machinery/field/generator/field_gen_2 = null
 
 /obj/machinery/field/containment/Initialize(mapload)
+	AddElement(/datum/element/blocks_explosives)
 	. = ..()
```

Comment:
Why does the addelement go before this?

Reply:
It's to pass unit tests, which I think occurs somewhere in the parent initialize calls. Can't detect the element if it was added after the parent class initialize calls.

Reply:
lol i looked at the code for obj/structure/window and it also does this

Reply:
I am so sorry